### PR TITLE
Namespace Abstraction 

### PR DESCRIPTION
**Pull request by andreitr**
TLDR: All properties, methods, and classes related to tenants are encapsulated within the Tenant class and are accessible through a read-only interface. Upon initialization, the Tenant class will generate contract, tokens, and UI configuration classes using relevant read-only factories.

Moving forward, tenant configuration should be accessed **exclusively** through the Tenant singleton, not via environment variables. Here's a quick diagram of how things work right now:

Untitled

## PR-Codex overview
This PR updates file imports and adds `Tenant` class for namespace handling. 

### Detailed summary
- Updated file imports to use `@/` alias
- Added `Tenant` class for namespace handling
- Refactored functions to use `Tenant.getInstance()` for namespace access

> The following files were skipped due to too many changes: `src/components/Delegates/DelegateCard/DelegateProfileImage.tsx`, `src/components/Metrics/DAOMetricsHeader.jsx`, `src/lib/tenantContract.ts`, `src/lib/tenant.ts`, `src/components/Proposals/ProposalPage/OPProposalApprovalPage/OptionResultsPanel/OptionResultsPanel.tsx`, `src/app/api/common/delegateStatement/createDelegateStatement.ts`, `src/components/Proposals/ProposalPage/OPProposalApprovalPage/OPProposalApprovalPage.tsx`, `src/app/api/common/proposals/getNeedsMyVoteProposals.ts`, `src/app/delegates/page.jsx`, `src/app/page.jsx`, `src/app/api/common/delegateStatement/getDelegateStatement.ts`, `src/lib/types.d.ts`, `src/lib/tokenUtils.ts`, `src/app/api/common/citizens/getCitizens.ts`, `src/app/api/votes/getVotes.ts`, `src/lib/tenantContractFactory.ts`, `src/app/api/delegations/getDelegations.ts`, `src/app/api/common/proposals/getProposals.ts`, `src/app/api/common/votes/getVotes.ts`, `src/app/api/common/delegates/getDelegates.ts`, `src/components/Proposals/ProposalPage/OPProposalPage/OPProposalOptimisticPage.jsx`, `src/app/api/common/delegations/getDelegations.ts`, `src/app/delegates/actions.ts`, `src/app/api/common/voting-power/getVotingPower.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

---

## PR-Codex overview
This PR focuses on restructuring file paths and introducing a `Tenant` class for namespace management.

### Detailed summary
- Moved functions to `getVotableSupply` and `getProposalTypes`
- Introduced `Tenant` class for namespace handling
- Updated file imports to use `@/` for common APIs

> The following files were skipped due to too many changes: `src/components/Delegates/DelegateCard/DelegateProfileImage.tsx`, `src/components/Metrics/DAOMetricsHeader.jsx`, `src/lib/tenantContract.ts`, `src/app/api/common/citizens/getCitizens.ts`, `src/lib/tenant.ts`, `src/app/api/common/delegateStatement/createDelegateStatement.ts`, `src/components/Proposals/ProposalPage/OPProposalApprovalPage/OptionResultsPanel/OptionResultsPanel.tsx`, `src/components/Proposals/ProposalPage/OPProposalApprovalPage/OPProposalApprovalPage.tsx`, `src/app/api/common/proposals/getNeedsMyVoteProposals.ts`, `src/app/delegates/page.jsx`, `src/app/page.jsx`, `src/app/api/common/delegateStatement/getDelegateStatement.ts`, `src/lib/types.d.ts`, `src/app/api/votes/getVotes.ts`, `src/lib/tenantContractFactory.ts`, `src/lib/tokenUtils.ts`, `src/app/api/delegations/getDelegations.ts`, `src/app/api/common/proposals/getProposals.ts`, `src/app/api/common/votes/getVotes.ts`, `src/app/api/common/delegates/getDelegates.ts`, `src/components/Proposals/ProposalPage/OPProposalPage/OPProposalOptimisticPage.jsx`, `src/app/api/common/delegations/getDelegations.ts`, `src/app/delegates/actions.ts`, `src/app/api/common/voting-power/getVotingPower.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

---

## PR-Codex overview
This PR updates import paths to use `@/` alias, adds `Tenant` class for namespace handling, and refactors functions for better modularity.

### Detailed summary
- Updated import paths to use `@/` alias
- Added `Tenant` class for namespace handling
- Refactored functions for better modularity

> The following files were skipped due to too many changes: `src/components/Delegates/DelegateCard/DelegateProfileImage.tsx`, `src/components/Metrics/DAOMetricsHeader.jsx`, `src/lib/tenantContract.ts`, `src/app/api/common/citizens/getCitizens.ts`, `src/lib/tenant.ts`, `src/app/api/common/delegateStatement/createDelegateStatement.ts`, `src/components/Proposals/ProposalPage/OPProposalApprovalPage/OptionResultsPanel/OptionResultsPanel.tsx`, `src/components/Proposals/ProposalPage/OPProposalApprovalPage/OPProposalApprovalPage.tsx`, `src/app/api/common/proposals/getNeedsMyVoteProposals.ts`, `src/app/delegates/page.jsx`, `src/app/page.jsx`, `src/app/api/common/delegateStatement/getDelegateStatement.ts`, `src/lib/types.d.ts`, `src/app/api/votes/getVotes.ts`, `src/lib/tenantContractFactory.ts`, `src/lib/tokenUtils.ts`, `src/app/api/delegations/getDelegations.ts`, `src/app/api/common/proposals/getProposals.ts`, `src/app/api/common/votes/getVotes.ts`, `src/app/api/common/delegates/getDelegates.ts`, `src/components/Proposals/ProposalPage/OPProposalPage/OPProposalOptimisticPage.jsx`, `src/app/api/common/delegations/getDelegations.ts`, `src/app/delegates/actions.ts`, `src/app/api/common/voting-power/getVotingPower.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kent** [comment]
Love the Tenant abstraction singleton pattern at work here, and love how the contract config can sit inside the files. 

My only feedback would be to further modularize the contract, and contract types to different files to keep this one nice and clean. Just thinking it will get quite big. But overall, love the direction!

File: src/app/api/common/delegations/getDelegations.ts [hunk 1]
```diff
@@ -5,31 +5,20 @@ import provider from "@/app/lib/provider";
...
-  addressOrENSName,
-  namespace,
-}: {
-  addressOrENSName: string;
-  namespace: "optimism";
-}) =>
-  addressOrEnsNameWrap(getCurrentDelegateesForAddress, addressOrENSName, {
-    namespace,
-  });
+export const getCurrentDelegatees = (addressOrENSName: string) =>
```
Comment (kent) [inline src/app/api/common/delegations/getDelegations.ts:14]: Nice

File: src/lib/tenant.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+export default class Tenant implements ITenant {
+  private static instance: Tenant;
+
+  private readonly _isProd: boolean;
+  private readonly _namespace: TenantNamespace;
+  private readonly _slug: string;
+  private readonly _token: TenantToken;
+
+  private constructor() {
```
Comment (kent) [inline src/lib/tenant.ts:20]: Do you think we will use DB or ENV here long term.

File: src/lib/tenantContractFactory.ts
Comment (kent) [inline src/lib/tenantContractFactory.ts:1]: Love this.

File: src/lib/tokenUtils.ts [hunk 3]
```diff
@@ -1,43 +1,19 @@
...
-  [
-    "optimism",
-    {
-      name: "Optimism",
-      symbol: "OP",
-      decimals: 18,
-    },
-  ],
-  [
-    "0x4200000000000000000000000000000000000042",
```
Comment (stepandel) [inline src/lib/tokenUtils.ts:20]: We still might need to resolve tokens by contract address when resolving proposals transaction data

File: src/lib/types.d.ts [hunk 4]
```diff
@@ -1,38 +1,54 @@
+import { type ITenantContract } from "@retro-pgf/contracts";
+
+export type TenantNamespace = "optimism";
+
+export type TenantContracts = {
+  alligator?: ITenantContract;
+  governor: ITenantContract;
+  proposalTypesConfigurator: ITenantContract;
```
Comment (stepandel) [inline src/lib/types.d.ts:8]: `proposalTypesConfigurator` is optional. At this point it's OP specific contract. All DAOs will only have `governor` and `token` as mandatory

**Review by stepandel — COMMENTED** [review]
Like the direction of this!

Left a few small comments. Besides that I think there’s a potential issue of opening up all api functions to all namespaces. Since not all of them can actually be accessed by all tenants, we going to need some default fallback or some sort of guard to make these functions inaccessible.
Eg. All advanced delegation functions

**Pull request merged**